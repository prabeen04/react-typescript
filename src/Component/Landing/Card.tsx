import * as React from 'react';
import { Link } from "react-router-dom";
// import { ICardData } from "./LandingInterface";

export default function Card(props: any) {
    const { title, description, action } = props.card;
    return (
        <div className='feature-card'>
            <h3>{title}</h3>
            <p>{description}</p>
            <Link to={action}>
                <button className="btn-danger btn text-btn">Explore &nbsp;<i className="fas fa-long-arrow-alt-right"></i></button>
            </Link>
        </div>
    )
}
