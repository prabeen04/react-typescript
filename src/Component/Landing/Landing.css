.landing-wrapper{
    background:linear-gradient(0deg,rgba(54, 4, 4, 0.925),rgba(20, 20, 20, 0.897)),url('../../Assets//Image/Landing.jpg');
    background-size:cover;
    height: calc(100vh - 71px);
}
.welcome-text{
    display:flex;
    justify-content: center;
    padding: 2rem;
}
.welcome-text > h1 {
    color: rgb(48, 47, 47);
    font-family: margarine;
}