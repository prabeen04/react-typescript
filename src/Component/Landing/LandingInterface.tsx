export interface ITitle {
    title: string;
}
export interface ICardData {
    title: string;
    description: string;
    action?: string | null;
}