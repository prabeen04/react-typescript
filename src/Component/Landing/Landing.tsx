import * as React from 'react';
import { useSpring, animated, useTransition } from 'react-spring';
import './Landing.css';

export default function Landing() {
    const styles = useSpring({ opacity: 1, transform: 'translate3d(0px,0,0) ', from: { opacity: 0, transform: 'translate3d(0,200px,0) ', } })
    const [items, set] = React.useState(['a', 'b', 'c', 'v'])
    const transitions = useTransition(items, item => item, {
        from: { transform: 'translate3d(0,-40px,0)', color: 'red' },
        enter: { transform: 'translate3d(0,0px,0)', color: 'blue'  },
        leave: { transform: 'translate3d(0,-40px,0)', color: 'yellow'  },
    })
    return (
        <div className='landing-wrapper'>
            <div className="welcome-text">
                <animated.h1 style={styles}>
                    <h1>Let's explore the POWER of typescript</h1>
                </animated.h1>
            </div>
            <div className="jumbotron">
                {
                    transitions.map(({ item, props, key }) =>
                        <animated.div key={key} style={props}>{item}</animated.div>
                    )
                }
            </div>
        </div>
    )
}
