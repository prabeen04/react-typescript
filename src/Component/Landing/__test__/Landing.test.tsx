import * as React from 'react';
import { shallow } from 'enzyme';
import Landing from '../Landing';

describe('<Landing />', () => {
    it('should render the Landing wrapper', () => {
        const wrapper = shallow(<Landing />);
        const landingWrapper = wrapper.find('[data-test="Landing"]');
        expect(landingWrapper.length).toEqual(1)
    })
})