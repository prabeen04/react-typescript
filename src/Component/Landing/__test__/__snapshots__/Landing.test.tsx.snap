// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="welcome-text"
      >
        <ForwardRef
          style={
            Object {
              "opacity": AnimatedValue {
                "animatedStyles": Set {},
                "children": Array [],
                "done": false,
                "lastPosition": 0,
                "lastTime": undefined,
                "lastVelocity": undefined,
                "payload": undefined,
                "setValue": [Function],
                "startPosition": 0,
                "startTime": 1554004685321,
                "value": 0,
              },
              "rotate": AnimatedInterpolation {
                "attach": [Function],
                "calc": [Function],
                "children": Array [],
                "detach": [Function],
                "payload": Array [
                  AnimatedValue {
                    "animatedStyles": Set {},
                    "children": Array [],
                    "done": false,
                    "lastPosition": 0,
                    "lastTime": undefined,
                    "lastVelocity": undefined,
                    "payload": undefined,
                    "setValue": [Function],
                    "startPosition": 0,
                    "startTime": 1554004685321,
                    "value": 0,
                  },
                ],
              },
              "transform": AnimatedInterpolation {
                "attach": [Function],
                "calc": [Function],
                "children": Array [],
                "detach": [Function],
                "payload": Array [
                  AnimatedValue {
                    "animatedStyles": Set {},
                    "children": Array [],
                    "done": false,
                    "lastPosition": 0,
                    "lastTime": undefined,
                    "lastVelocity": undefined,
                    "payload": undefined,
                    "setValue": [Function],
                    "startPosition": 0,
                    "startTime": 1554004685321,
                    "value": 0,
                  },
                ],
              },
            }
          }
        >
          <React.Fragment>
            <h1>
              Let's explore the
              <Title
                title=" POWER "
              />
              of typescript
            </h1>
          </React.Fragment>
        </ForwardRef>
      </div>,
      "className": "landing-wrapper",
      "data-test": "Landing",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef
          style={
            Object {
              "opacity": AnimatedValue {
                "animatedStyles": Set {},
                "children": Array [],
                "done": false,
                "lastPosition": 0,
                "lastTime": undefined,
                "lastVelocity": undefined,
                "payload": undefined,
                "setValue": [Function],
                "startPosition": 0,
                "startTime": 1554004685321,
                "value": 0,
              },
              "rotate": AnimatedInterpolation {
                "attach": [Function],
                "calc": [Function],
                "children": Array [],
                "detach": [Function],
                "payload": Array [
                  AnimatedValue {
                    "animatedStyles": Set {},
                    "children": Array [],
                    "done": false,
                    "lastPosition": 0,
                    "lastTime": undefined,
                    "lastVelocity": undefined,
                    "payload": undefined,
                    "setValue": [Function],
                    "startPosition": 0,
                    "startTime": 1554004685321,
                    "value": 0,
                  },
                ],
              },
              "transform": AnimatedInterpolation {
                "attach": [Function],
                "calc": [Function],
                "children": Array [],
                "detach": [Function],
                "payload": Array [
                  AnimatedValue {
                    "animatedStyles": Set {},
                    "children": Array [],
                    "done": false,
                    "lastPosition": 0,
                    "lastTime": undefined,
                    "lastVelocity": undefined,
                    "payload": undefined,
                    "setValue": [Function],
                    "startPosition": 0,
                    "startTime": 1554004685321,
                    "value": 0,
                  },
                ],
              },
            }
          }
        >
          <React.Fragment>
            <h1>
              Let's explore the
              <Title
                title=" POWER "
              />
              of typescript
            </h1>
          </React.Fragment>
        </ForwardRef>,
        "className": "welcome-text",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <React.Fragment>
            <h1>
              Let's explore the
              <Title
                title=" POWER "
              />
              of typescript
            </h1>
          </React.Fragment>,
          "style": Object {
            "opacity": AnimatedValue {
              "animatedStyles": Set {},
              "children": Array [],
              "done": false,
              "lastPosition": 0,
              "lastTime": undefined,
              "lastVelocity": undefined,
              "payload": undefined,
              "setValue": [Function],
              "startPosition": 0,
              "startTime": 1554004685321,
              "value": 0,
            },
            "rotate": AnimatedInterpolation {
              "attach": [Function],
              "calc": [Function],
              "children": Array [],
              "detach": [Function],
              "payload": Array [
                AnimatedValue {
                  "animatedStyles": Set {},
                  "children": Array [],
                  "done": false,
                  "lastPosition": 0,
                  "lastTime": undefined,
                  "lastVelocity": undefined,
                  "payload": undefined,
                  "setValue": [Function],
                  "startPosition": 0,
                  "startTime": 1554004685321,
                  "value": 0,
                },
              ],
            },
            "transform": AnimatedInterpolation {
              "attach": [Function],
              "calc": [Function],
              "children": Array [],
              "detach": [Function],
              "payload": Array [
                AnimatedValue {
                  "animatedStyles": Set {},
                  "children": Array [],
                  "done": false,
                  "lastPosition": 0,
                  "lastTime": undefined,
                  "lastVelocity": undefined,
                  "payload": undefined,
                  "setValue": [Function],
                  "startPosition": 0,
                  "startTime": 1554004685321,
                  "value": 0,
                },
              ],
            },
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h1>
              Let's explore the
              <Title
                title=" POWER "
              />
              of typescript
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Let's explore the",
                <Title
                  title=" POWER "
                />,
                "of typescript",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Let's explore the",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "title": " POWER ",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "of typescript",
            ],
            "type": "h1",
          },
          "type": Symbol(react.fragment),
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="welcome-text"
        >
          <ForwardRef
            style={
              Object {
                "opacity": AnimatedValue {
                  "animatedStyles": Set {},
                  "children": Array [],
                  "done": false,
                  "lastPosition": 0,
                  "lastTime": undefined,
                  "lastVelocity": undefined,
                  "payload": undefined,
                  "setValue": [Function],
                  "startPosition": 0,
                  "startTime": 1554004685321,
                  "value": 0,
                },
                "rotate": AnimatedInterpolation {
                  "attach": [Function],
                  "calc": [Function],
                  "children": Array [],
                  "detach": [Function],
                  "payload": Array [
                    AnimatedValue {
                      "animatedStyles": Set {},
                      "children": Array [],
                      "done": false,
                      "lastPosition": 0,
                      "lastTime": undefined,
                      "lastVelocity": undefined,
                      "payload": undefined,
                      "setValue": [Function],
                      "startPosition": 0,
                      "startTime": 1554004685321,
                      "value": 0,
                    },
                  ],
                },
                "transform": AnimatedInterpolation {
                  "attach": [Function],
                  "calc": [Function],
                  "children": Array [],
                  "detach": [Function],
                  "payload": Array [
                    AnimatedValue {
                      "animatedStyles": Set {},
                      "children": Array [],
                      "done": false,
                      "lastPosition": 0,
                      "lastTime": undefined,
                      "lastVelocity": undefined,
                      "payload": undefined,
                      "setValue": [Function],
                      "startPosition": 0,
                      "startTime": 1554004685321,
                      "value": 0,
                    },
                  ],
                },
              }
            }
          >
            <React.Fragment>
              <h1>
                Let's explore the
                <Title
                  title=" POWER "
                />
                of typescript
              </h1>
            </React.Fragment>
          </ForwardRef>
        </div>,
        "className": "landing-wrapper",
        "data-test": "Landing",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ForwardRef
            style={
              Object {
                "opacity": AnimatedValue {
                  "animatedStyles": Set {},
                  "children": Array [],
                  "done": false,
                  "lastPosition": 0,
                  "lastTime": undefined,
                  "lastVelocity": undefined,
                  "payload": undefined,
                  "setValue": [Function],
                  "startPosition": 0,
                  "startTime": 1554004685321,
                  "value": 0,
                },
                "rotate": AnimatedInterpolation {
                  "attach": [Function],
                  "calc": [Function],
                  "children": Array [],
                  "detach": [Function],
                  "payload": Array [
                    AnimatedValue {
                      "animatedStyles": Set {},
                      "children": Array [],
                      "done": false,
                      "lastPosition": 0,
                      "lastTime": undefined,
                      "lastVelocity": undefined,
                      "payload": undefined,
                      "setValue": [Function],
                      "startPosition": 0,
                      "startTime": 1554004685321,
                      "value": 0,
                    },
                  ],
                },
                "transform": AnimatedInterpolation {
                  "attach": [Function],
                  "calc": [Function],
                  "children": Array [],
                  "detach": [Function],
                  "payload": Array [
                    AnimatedValue {
                      "animatedStyles": Set {},
                      "children": Array [],
                      "done": false,
                      "lastPosition": 0,
                      "lastTime": undefined,
                      "lastVelocity": undefined,
                      "payload": undefined,
                      "setValue": [Function],
                      "startPosition": 0,
                      "startTime": 1554004685321,
                      "value": 0,
                    },
                  ],
                },
              }
            }
          >
            <React.Fragment>
              <h1>
                Let's explore the
                <Title
                  title=" POWER "
                />
                of typescript
              </h1>
            </React.Fragment>
          </ForwardRef>,
          "className": "welcome-text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <React.Fragment>
              <h1>
                Let's explore the
                <Title
                  title=" POWER "
                />
                of typescript
              </h1>
            </React.Fragment>,
            "style": Object {
              "opacity": AnimatedValue {
                "animatedStyles": Set {},
                "children": Array [],
                "done": false,
                "lastPosition": 0,
                "lastTime": undefined,
                "lastVelocity": undefined,
                "payload": undefined,
                "setValue": [Function],
                "startPosition": 0,
                "startTime": 1554004685321,
                "value": 0,
              },
              "rotate": AnimatedInterpolation {
                "attach": [Function],
                "calc": [Function],
                "children": Array [],
                "detach": [Function],
                "payload": Array [
                  AnimatedValue {
                    "animatedStyles": Set {},
                    "children": Array [],
                    "done": false,
                    "lastPosition": 0,
                    "lastTime": undefined,
                    "lastVelocity": undefined,
                    "payload": undefined,
                    "setValue": [Function],
                    "startPosition": 0,
                    "startTime": 1554004685321,
                    "value": 0,
                  },
                ],
              },
              "transform": AnimatedInterpolation {
                "attach": [Function],
                "calc": [Function],
                "children": Array [],
                "detach": [Function],
                "payload": Array [
                  AnimatedValue {
                    "animatedStyles": Set {},
                    "children": Array [],
                    "done": false,
                    "lastPosition": 0,
                    "lastTime": undefined,
                    "lastVelocity": undefined,
                    "payload": undefined,
                    "setValue": [Function],
                    "startPosition": 0,
                    "startTime": 1554004685321,
                    "value": 0,
                  },
                ],
              },
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h1>
                Let's explore the
                <Title
                  title=" POWER "
                />
                of typescript
              </h1>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Let's explore the",
                  <Title
                    title=" POWER "
                  />,
                  "of typescript",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Let's explore the",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "title": " POWER ",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "of typescript",
              ],
              "type": "h1",
            },
            "type": Symbol(react.fragment),
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
